### PR TITLE
Improving wording related to Boost library requirements

### PR DESCRIPTION
**Pull request by jonathancross**
Documentation was unclear in this section and could be interpreted to mean that boost was not a hard requirement for older Ubuntu versions.
Related: https://github.com/bitcoin/bitcoin/issues/7587

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -67,15 +67,18 @@ Build requirements:
...
-On at least Ubuntu 14.04+ and Debian 7+ there are generic names for the
+
+Options when installing required Boost library files:
+
+1. On at least Ubuntu 14.04+ and Debian 7+ there are generic names for the
 individual boost development packages, so the following can be used to only
 install necessary parts of boost:
 
-    sudo apt-get install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev
+    `sudo apt-get install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev`
```
Comment (jonasschnelli) [inline doc/build-unix.md]: nit: IMO if you use a 4-spaces-indent that's equivalent to \`

**Comment by jonasschnelli** [comment]
ACK 417e64bc0176db5d17d8ed0f03fc64154cbd00b8

File: doc/build-unix.md (hunk 1 above)
Comment (laanwj) [inline doc/build-unix.md]: Yes, either do

    install necessary parts of boost:
    ```bash
    <commands go here>
    ```

Or indent four spaces:

    install necessary parts of boost:

        <commands go here>

Both is not valid markdown.

Comment (jonathancross) [inline doc/build-unix.md]: Thanks Jonas.
The Ordered list (numbers 1 and 2) formatting cause markdown to ignore the 4-space "code" indentation markup.  I did [some experimentation](https://github.com/jonathancross/my-docs/blob/master/Github%20markdown%20list%20and%20code.md) and now see that I need to indent 8 spaces to bring back the code formatting when in a list.

File: doc/build-unix.md [hunk 2]
```diff
@@ -67,15 +67,18 @@ Build requirements:
 
     sudo apt-get install build-essential libtool autotools-dev automake pkg-config libssl-dev libevent-dev bsdmainutils
 
-On at least Ubuntu 14.04+ and Debian 7+ there are generic names for the
+
+Options when installing required Boost library files:
+
+1. On at least Ubuntu 14.04+ and Debian 7+ there are generic names for the
```
Comment (MarcoFalke) [inline doc/build-unix.md]: I am not sure but this may be Debian 8+ as implied here: 197a7a8c7ace9ff54010adb836b4e7e35b8ead18

**Comment by jonathancross** [comment]
I went ahead and created a new clean pull request #7590 with a single commit and the formatting chances requested.

**Pull request closed without merging**